test/python/quantum_info/operators/symplectic/test_clifford.py

```diff
@@ -989,13 +1037,58 @@ def test_instruction_name(self, num_qubits):
         clifford = random_clifford(num_qubits, seed=777)
         self.assertEqual(clifford.to_instruction().name, str(clifford))
 
-    def visualize_does_not_throw_error(self):
+    def test_visualize_does_not_throw_error(self):
         """Test to verify that drawing Clifford does not throw an error"""
         # An error may be thrown if visualization code calls op.condition instead
         # of getattr(op, "condition", None)
         clifford = random_clifford(3, seed=0)
         print(clifford)
 
+    @combine(num_qubits=[1, 2, 3, 4])
+    def test_from_matrix_round_trip(self, num_qubits):
+        """Test round trip conversion to and from matrix"""
+        for i in range(10):
+            expected = random_clifford(num_qubits, seed=42 + i)
+            actual = Clifford.from_matrix(expected.to_matrix())
+            self.assertEqual(expected, actual)
+
+    @combine(num_qubits=[1, 2, 3, 4])
+    def test_from_operator_round_trip(self, num_qubits):
+        """Test round trip conversion to and from operator"""
+        for i in range(10):
+            expected = random_clifford(num_qubits, seed=777 + i)
+            actual = Clifford.from_operator(expected.to_operator())
+            self.assertEqual(expected, actual)
+
+    @combine(
+        gate=[
```

Comment:
Perhaps it's worth to add a test also for the inverse of these gates, e.g. check that it's possible to create a Clifford also from `ECRGate().inverse()` and similarly also to the other gates in the list below.

Reply:
I think the issue on `inverse` is an issue on `Gate` rather than `Clifford`. So it would be better to create a new issue on `Gate.inverse` rather than adding test cases for that here and fixing it within this PR.

Anyway, I see your point. I'll update the release note to be clear that using `from_matrix` may not cover all the cases unless `to_matrix` is provided for all operations. In fact, currently `ECRGate.inverse()` and `iSwapGate.inverse()` do not provide `to_matrix` and raise an error:
```
qiskit.circuit.exceptions.CircuitError: "to_matrix not defined for this <class 'qiskit.circuit.gate.Gate'>"
```

It may be possible to add `to_matrix` for `ECRGate.inverse()` but I'm not sure we should do that since people may not want faster `inverse` without `to_matrix` rather than slower `inverse` with `to_matrix`. Simply having `ECRdgGate` and `iSwapdgGate` may not be so bad. Also, I've heard Sasha is working on something related with the issue on `Gate.inverse`. Such things should be discussed in the new issue.